docs/type_promotion.rst

```diff
@@ -5,14 +5,42 @@ Type promotion semantics
 
 JAX's type promotion rules (i.e., the result of
 :func:`jax.numpy.promote_types` for each pair of types) are given by the
-following table, where, for example
+following type promotion lattice, where, for example:
 
 * "b1" means :code:`np.bool_`,
-* "s2" means :code:`np.int16`,
+* "i2" means :code:`np.int16`,
 * "u4" means :code:`np.uint32`,
 * "bf" means :code:`np.bfloat16`,
-* "f2" means :code:`np.float16`, and
-* "c8" means :code:`np.complex128`.
+* "f2" means :code:`np.float16`,
+* "c8" means :code:`np.complex128`,
+* "i*" means Python :code:`int`,
```

Comment:
Perhaps we can use i_ like in the code?

Reply:
I prefer `i*` because using the standard wildcard character communicates that the precision is variable.